### PR TITLE
Add Title and Body Displays to Block Content View

### PR DESCRIPTION
**Pull request by HemanshuGandhi**
As per Hemanshu's deliverables in #380. 

Screen Shot 2020-04-12 at 4 14 59 AM

### CODE REVIEW

File: src/main/java/com/notably/view/blockcontent/BlockContentDisplayView.java [hunk 1]
```diff
@@ -18,48 +18,82 @@
...
+    private void setBlockContentDisplay(Model model) {
         Objects.requireNonNull(model);
 
         AbsolutePath currentlyOpenPath = model.getCurrentlyOpenPath();
         BlockTreeItem currentlyOpenBlock = model.getBlockTree().get(currentlyOpenPath);
-        String markdownBody = currentlyOpenBlock.getBody().getText();
 
+        String htmlTitle = getHtmlTitle(currentlyOpenBlock);
+        String htmlBody = getHtmlBody(currentlyOpenBlock);
+        String htmlContent = htmlTitle + NEWLINE + htmlBody;
```
Comment (kevinputera) [inline src/main/java/com/notably/view/blockcontent/BlockContentDisplayView.java]: Actually, HTML doesn't require respect newlines. You can remove this newline and everything will still work fine!

**Review by kevinputera — COMMENTED** [review]
Hey @HemanshuGandhi , looks great! Some comments:

1. How'd you plan to accommodate for this (if the body contains an \<h1>)?
image
2. Do you have plans for including the path string beside the title (as drawn in the mockup)?
3. Currently, the backend do not support editing a block's title. Should we just remove that functionality from `BlockContentEditView`?

File: src/main/java/com/notably/view/blockcontent/BlockContentDisplayView.java [hunk 2]
```diff
@@ -21,47 +28,126 @@
...
-                .addEventHandler(TreeItem.treeNotificationEvent(), event -> reload(model));
+    private void setSettings() {
+        blockPathDisplay.setTextAlignment(TextAlignment.RIGHT);
+        blockTitlePathContainer.setMinWidth(50);
+        blockTitlePathContainer.setMinWidth(0);
+        blockTitlePathContainer.setPrefWidth(1);
+        blockTitlePathContainer.setAlignment(Pos.CENTER_LEFT);
+        blockTitlePathContainer.setHgrow(blockPathDisplay, Priority.SOMETIMES);
+        blockTitlePathContainer.setHgrow(separator, Priority.SOMETIMES);
+        blockTitlePathContainer.setHgrow(blockTitleDisplay, Priority.ALWAYS);
```
Comment (kevinputera) [inline src/main/java/com/notably/view/blockcontent/BlockContentDisplayView.java]: Nitpicking here but I believe the recommended way of calling static methods is `<ClassName>.<static method name>`. For that reason, this should be `HBox.setHgrow`. You can actually set all these parameters directly in the `fxml` files too!

**Review by kevinputera — APPROVED** [review]
LGTM!

File: src/main/java/com/notably/view/blockcontent/BlockContentDisplayView.java (hunk 2 above)
Comment (HemanshuGandhi) [reply src/main/java/com/notably/view/blockcontent/BlockContentDisplayView.java]: I've converted the inline stylings to fxml. Thanks.

**Review by kevinputera — APPROVED** [review]
LGTM!

**Pull request merged**